arch/x86/kernel/apic/apic.c

```diff
@@ -324,6 +324,19 @@ int lapic_get_maxlvt(void)
 /* i82489DX specific */
 #define		I82489DX_BASE_DIVIDER		(((0x2) << 18))
 
+/* Now QNX uses Core Base Frequency instead of nominal core crystal clock.
+ * We need to scale LVTT Initial Count by TSC/clock ratio. On RPL, it's 73 (2800MHZ / 38400000 Hz)
+ */
+static int qnx_lvtt_scale_factor = 1;
+
+static int __init qnx_lvtt_scale_factor_setup(char *str)
+{
+	if (hypervisor_is_type(X86_HYPER_QNX))
+		get_option(&str, &qnx_lvtt_scale_factor);
+	return 1;
```

Comment:
If we are not relying on any of the return value from function, why to return hardcoded 1?
Can we not make it void?

Reply:
This is a callback for linux __setup macro.

/*
 * NOTE: __setup functions return values:
 * @fn returns 1 (or non-zero) if the option argument is "handled"
 * and returns 0 if the option argument is "not handled".
 */
#define __setup(str, fn)						\
	__setup_param(str, fn, fn, 0)

So i think it should return a value.

Reply:
Can we put a comment like callback function. are you sure you always want to return 1 from this function?